### PR TITLE
fix various problems surfaced based on migrating real data

### PR DESCRIPTION
**Pull request by Pomax**
- fixed vote copying in migrate_products
- template update to make creep-o-meter work again (`data-creepiness`)
- fixed `review_date` field, so that the `catalog.html` template can properly filter on `product.is_current`
- updated productpage factory code to include a superclass productpagefactory that covers the shared fields
  - includes vote faking so we have a spread of creepiness

review app:

- https://foundation-s-fix-produc-yps4gz.herokuapp.com/cms/pages/89/
- https://foundation-s-fix-produc-yps4gz.herokuapp.com/en/privacynotincluded-new

### CODE REVIEW

**Review by KalobTaulien — APPROVED** [review]

File: network-api/networkapi/wagtailpages/pagemodels/products.py [hunk 1]
```diff
@@ -184,7 +185,7 @@ class ProductPage(FoundationMetadataPageMixin, Page):
     )
     review_date = models.DateField(
         help_text='Review date of this product',
-        auto_now_add=True,
+        default=timezone.now
```
Comment (Pomax) [inline network-api/networkapi/wagtailpages/pagemodels/products.py:188]: by using `auto_now_add` this field had become read-only, which meant we couldn't actually copy over values from real products. This should do the same thing, but allows the field to be set to whatever we need.

File: network-api/networkapi/wagtailpages/pagemodels/products.py [hunk 2]
```diff
@@ -342,15 +343,15 @@ class ProductPage(FoundationMetadataPageMixin, Page):
     )
 
     @property
-    def current_tally(self):
+    def total_vote_count(self):
```
Comment (Pomax) [inline network-api/networkapi/wagtailpages/pagemodels/products.py:346]: renamed to reflect its role

File: network-api/networkapi/wagtailpages/pagemodels/products.py [hunk 3]
```diff
@@ -342,15 +343,15 @@ class ProductPage(FoundationMetadataPageMixin, Page):
...
         return sum(self.get_or_create_votes())
 
     @property
     def creepiness(self):
         try:
-            average = self.creepiness_value / self.current_tally
+            average = self.creepiness_value / self.total_vote_count
         except ZeroDivisionError:
-            average = 0
+            average = 50
```
Comment (Pomax) [inline network-api/networkapi/wagtailpages/pagemodels/products.py:354]: based on the code in use by PNI right now

File: network-api/networkapi/wagtailpages/pagemodels/products.py [hunk 4]
```diff
@@ -364,16 +365,18 @@ def get_voting_json(self):
...
         return json.dumps(data)
 
     content_panels = Page.content_panels + [
         MultiFieldPanel(
             [
+                FieldPanel('review_date'),
                 FieldPanel('privacy_ding'),
                 FieldPanel('adult_content'),
                 FieldPanel('company'),
+                FieldPanel('company'),
```
Comment (Pomax) [inline network-api/networkapi/wagtailpages/pagemodels/products.py:379]:
```suggestion
```

File: network-api/networkapi/buyersguide/management/commands/migrate_products.py [hunk 5]
```diff
@@ -187,10 +187,13 @@ def handle(self, *args, **options):
                 new_product_page.get_or_create_votes()
                 # Use .to_dict() to pull out the old aggregated votes
                 product_dict = product.to_dict()
-                if hasattr(product_dict, 'votes'):
-                    votes = product_dict['votes']['creepiness']['vote_breakdown']
+                votes = product_dict.get('votes', None)
+                if votes:
+                    print(votes)
```
Comment (KalobTaulien) [inline network-api/networkapi/buyersguide/management/commands/migrate_products.py:192]: Let's use `self.debug_print()` here so we don't spam the production or staging logs.

File: network-api/networkapi/wagtailpages/migrations/0023_auto_20201204_0105.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+    dependencies = [
+        ('wagtailpages', '0022_product_voting'),
+    ]
+
+    operations = [
+        migrations.AlterField(
+            model_name='productpage',
+            name='review_date',
+            field=models.DateField(default=django.utils.timezone.now, help_text='Review date of this product'),
```
Comment (KalobTaulien) [inline network-api/networkapi/wagtailpages/migrations/0023_auto_20201204_0105.py:17]: Could we put this `migrations.AlertField` inside of `0022_product_voting.py`? One less migration file

File: network-api/networkapi/buyersguide/templates/buyersguide/catalog.html [hunk 7]
```diff
@@ -67,9 +67,9 @@ <h1 class="h1-heading text-center mt-0">{% trans "Be Smart. Shop Safe." %}</h1>
     <div class="product-box-list d-flex justify-content-center align-items-stretch flex-wrap">
 
       {% for product in products %}
-        {% if product.is_current %}
+        {% if product.specific.is_current %}
```
Comment (KalobTaulien) [inline network-api/networkapi/buyersguide/templates/buyersguide/catalog.html:70]: Not sure why (I haven't dug into it) but the products don't show up on the BuyersGuidePage when I run `migrate_products`. I suspect it has to do with this line. 

Edit: Confirmed, it's this line preventing products from showing up after the `migrate_products` script is run.

**Review by KalobTaulien — CHANGES_REQUESTED** [review]

File: network-api/networkapi/buyersguide/management/commands/migrate_products.py (hunk 5 above)
Comment (Pomax) [reply network-api/networkapi/buyersguide/management/commands/migrate_products.py:192]:
```suggestion
                    self.debug_print(votes)
```

File: network-api/networkapi/wagtailpages/migrations/0023_auto_20201204_0105.py (hunk 6 above)
Comment (Pomax) [reply network-api/networkapi/wagtailpages/migrations/0023_auto_20201204_0105.py:17]: +1

File: network-api/networkapi/buyersguide/templates/buyersguide/catalog.html (hunk 7 above)
Comment (Pomax) [reply network-api/networkapi/buyersguide/templates/buyersguide/catalog.html:70]: I still see products, just not "all products" - it should only show 136 of them

File: network-api/networkapi/buyersguide/factory.py [hunk 8]
```diff
@@ -89,8 +90,14 @@ class Meta:
...
-    review_date = Faker('date_time_between_dates',
-                        datetime_start=date(year=2020, month=11, day=1), datetime_end=None, tzinfo=timezone.utc)
+
+    review_date = Faker(
+        'date_time_between_dates',
+        datetime_start=date(year=2020, month=11, day=1),
+        datetime_end=None,
+        tzinfo=timezone.utc
+    )
+
```
Comment (Pomax) [inline network-api/networkapi/buyersguide/factory.py:100]:
```suggestion
review_date = Faker('date_time_between_dates',
                        datetime_start=date(year=2020, month=11, day=1), datetime_end=None, tzinfo=timezone.utc)
```

File: network-api/networkapi/buyersguide/factory.py [hunk 9]
```diff
@@ -89,7 +90,7 @@ class Meta:
     adult_content = Faker('boolean')
     uses_wifi = Faker('boolean')
     uses_bluetooth = Faker('boolean')
-    review_date = Faker('date_time_between_dates',
+review_date = Faker('date_time_between_dates',
```
Comment (Pomax) [inline network-api/networkapi/buyersguide/factory.py:93]:
```suggestion
    review_date = Faker('date_time_between_dates',
```

File: network-api/networkapi/wagtailpages/pagemodels/products.py [hunk 10]
```diff
@@ -184,7 +185,8 @@ class ProductPage(FoundationMetadataPageMixin, Page):
     )
     review_date = models.DateField(
         help_text='Review date of this product',
-        auto_now_add=True,
+        default=timezone.now,
+        null=True,
```
Comment (Pomax) [inline network-api/networkapi/wagtailpages/pagemodels/products.py:189]:
```suggestion
        default=timezone.now
```

Comment (Pomax) [inline network-api/networkapi/wagtailpages/pagemodels/products.py:189]:
```suggestion
        default=timezone.now,
```

File: network-api/networkapi/wagtailpages/pagemodels/products.py [hunk 11]
```diff
@@ -184,7 +185,8 @@ class ProductPage(FoundationMetadataPageMixin, Page):
...
     )
```
Comment (Pomax) [inline network-api/networkapi/wagtailpages/pagemodels/products.py:190]:
```suggestion
    )
```

File: network-api/networkapi/buyersguide/factory.py [hunk 12]
```diff
@@ -219,38 +220,63 @@ class Meta:
...
+    company = Faker('company')
+    blurb = Faker('sentence')
+    product_url = Faker('url')
+    price = LazyAttribute(lambda _: randint(49, 1500))
+    worst_case = Faker('sentence')
+
     first_published_at = Faker('past_datetime', start_date='-2d', tzinfo=timezone.utc)
     last_published_at = Faker('past_datetime', start_date='-1d', tzinfo=timezone.utc)
 
+    # TODO: add image binding
```
Comment (KalobTaulien) [inline network-api/networkapi/buyersguide/factory.py:253]: Should we add images in at this time? We landed some work that fixes the images in the template, but adding them here in the factory would be amazing. This is non blocking imo as it's just factory code, so if we don't add it in now let's make sure we open a ticket for it.

**Review by KalobTaulien — APPROVED** [review]
Everything seems to working as expected as far as I can tell. Code looks good! Just the one [optional] comment

File: network-api/networkapi/buyersguide/factory.py (hunk 12 above)
Comment (Pomax) [reply network-api/networkapi/buyersguide/factory.py:253]: let's make that part of https://github.com/mozilla/foundation.mozilla.org/issues/5777

**Pull request merged**